### PR TITLE
Dialogbox buttons: change the position of the boutons

### PR DESCRIPTION
**Pull request by HFaaaaa**
Ticket #79 
After trying to change the text content and padding, there are still some small screens that can’t adapt. Like that:
image
So I try the second proposition. Change the position of the button, change the padding of the dialog box and change the padding of the part buttons.
image

### CODE REVIEW

**Review by Jogo27 — CHANGES_REQUESTED** [review]
Thank you for the PR. The idea to display the button vertically when the screen is narrow is very good!
Please take the following comments into account though, in particular the requirement that **when the screen is large** the buttons must be displayed horizontally and right-aligned. Moreover, the design has to improve:
- The margin around the bock of button is too large, in particular below.
- There must be some space between the buttons (the same space as when the buttons are displayed horizontally)

File: app/src/styles.sass [hunk 1]
```diff
@@ -186,3 +186,35 @@ div.mat-menu-panel.nav-menu
...
+      border: solid 1px #602c57
+      border-radius: 3px
+      padding: 4px
+      color: #381a33
+      font-family: "Asap"
+      font-size: 12pt
+      background-color: #d5b6d0
+      font-size: 13pt
+      padding: 0.4em 1em
+      margin-left: 8px
```
[the remark this answers is not in the document]
Comment (HFaaaaa) [reply app/src/styles.sass:220]: Thanks for your comments!!! I will resubmit after modifying those problems.

**Comment by HFaaaaa** [comment]
PTAL

File: app/src/sizes.sass [hunk 2]
```diff
@@ -3,3 +3,6 @@ $navbarheight: 40px
 
 $small-space: 4px
 $medium-space: 8px
+
+
+$big-space: 45px
```
Comment (Jogo27) [inline app/src/sizes.sass:8]: $small-space and $medium-space are mostly spaces between borders or lines. If the font sizes change these spaces should remain the same. But if the border widths change then these spaces may change. On the contrary, it seems to me that the space between the dialog message and the button bar must be smaller if the font size is smaller and bigger if the font is bigger. Therefore, the size of this space should be written in ex or em unit. Moreover, I'm not convinced that the exact same space should be reused throughout the application. Hence I don't think it's a good candidate for an application-wide variable.

**Review by Jogo27 — CHANGES_REQUESTED** [review]
Dialog boxes look much better now, thanks! It could be even better though if all the buttons had the same width when displayed vertically. Please also take the following comments about your code into account.

**Comment by HFaaaaa** [comment]
PTAL

File: app/src/app/create/leave-create.dialog.html [hunk 3]
```diff
@@ -3,6 +3,7 @@
   What do you want to do?
 </div>
 <div class="formactions">
+
```
Comment (Jogo27) [inline app/src/app/create/leave-create.dialog.html:6]: Please remove this misleading empty line.

File: app/src/styles.sass [hunk 4]
```diff
@@ -42,8 +43,19 @@ input
     &[disabled], &[disabled]:hover
       border-color: tame($maincolor)
       background-color: tame($userbuttonbgcolor)
-      color: tame($useboxfgcolor)
+      color: tame($useboxfgcolor) 
 
+@media screen and (max-width: $breakpoint-tablet)
+  .dialog-box .formactions
+    padding: 0
+    display: flex
```
Comment (Jogo27) [inline app/src/styles.sass:51]: This line is useless. formactions are always flex.

File: app/src/styles.sass [hunk 5]
```diff
@@ -42,8 +43,19 @@ input
...
+    justify-content: center
```
Comment (Jogo27) [inline app/src/styles.sass:52]: This line is useless, because there is no external constraint on the height of the element.

**Review by Jogo27 — CHANGES_REQUESTED** [review]
It's almost Ok, now. In addition to the following comments could you please:
1. Remove the left margin on the buttons for them to really be aligned.
2. Give the same width to all the buttons.

**Comment by HFaaaaa** [comment]
PTAL

File: app/src/styles.sass [hunk 6]
```diff
@@ -42,8 +43,18 @@ input
...
+      color: tame($useboxfgcolor) 
 
+@media screen and (max-width: $breakpoint-tablet)
+  .dialog-box .formactions
+    padding: 0
+    flex-direction: column
+    align-items: stretch
+    margin-left: 0
+    button
+      @include userbox
```
Comment (Jogo27) [inline app/src/styles.sass:55]: You don't need to repeat that here. buttons in element with formactions class have already included userbox. Moreover, it results in errors, like the font size which is smaller on narrow screens. Just remove this line.

File: app/src/styles.sass [hunk 7]
```diff
@@ -42,8 +43,18 @@ input
...
       background-color: tame($userbuttonbgcolor)
-      color: tame($useboxfgcolor)
+      color: tame($useboxfgcolor) 
 
+@media screen and (max-width: $breakpoint-tablet)
+  .dialog-box .formactions
+    padding: 0
+    flex-direction: column
+    align-items: stretch
+    margin-left: 0
```
Comment (Jogo27) [inline app/src/styles.sass:53]: Useless. The left margin already is zero for formactions.

**Review by Jogo27 — CHANGES_REQUESTED** [review]
You still has to remove the left margin of buttons, and to take the following two remarks into account.

**Comment by HFaaaaa** [comment]
PTAL

**Review by Jogo27 — APPROVED** [review]

**Pull request merged**